### PR TITLE
Feature path std io

### PR DESCRIPTION
**Pull request by mquillot**
## What does this PR do?

Add support for "-" as value for Path class initialization so that user
can ask to use standard input/output instead of file.

For instance, users who are used to type their input files with Path_fr will now be able to give "-" as input string so that get_content or open (of Path_fr class) will read the standard input stream without having nothing more to do. Same for output.

Note that some code has been added for typehiting so that it does not create conflict with another feature: giving "-" to specify that a list of files is given through the standard input.

## Before submitting

- [ X ] Did you read the [contributing guideline](https://github.com/omni-us/jsonargparse/blob/main/CONTRIBUTING.rst)?
- [ X ] Did you update **the documentation**? (readme and public docstrings)
- [ X ] Did you write **unit tests** such that there is 100% coverage on related code? (required for bug fixes and new features)
- [ X ] Did you verify that new and existing **tests pass locally**?
- [ X ] Did you make sure that all changes preserve **backward compatibility**?
- [ X ] Did you update **the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

### CODE REVIEW

File: jsonargparse_tests/test_util.py [hunk 1]
```diff
@@ -677,3 +678,50 @@ def parse_args(args=[]):
...
+        with path.open("w") as std_output:
+            std_output.write("test\n")
+    assert path.std_io
+
+    path = Path("-", mode="w")
+    assert path == "-"
+    with patch("sys.stdout", StringIO("")):
+        with path.open("w") as std_output:
+            std_output.write("test\n")
+    assert path.std_io
```
Comment (mauvilsa) [inline jsonargparse_tests/test_util.py:720]: Remove these. No point with empty mode or with just "w".

File: jsonargparse_tests/test_util.py [hunk 2]
```diff
@@ -677,3 +678,50 @@ def parse_args(args=[]):
...
+
+    path = Path("-", mode="fw")
+    assert path == "-"
+    with patch("sys.stdout", StringIO("")):
```
Comment (mauvilsa) [inline jsonargparse_tests/test_util.py:724]: Save the `StringIO` instance to a variable, and after the write, assert that the content of the `StringIO` is what got printed.

**Review by mauvilsa — CHANGES_REQUESTED** [review]
Overall it looks very good. Thank you! Thought I do want some changes.

**Comment by mquillot** [comment]
I think @mauvilsa you can review a second time. I tried to follow you recommandations as soon as possible.

File: jsonargparse/_typehints.py [hunk 3]
```diff
@@ -501,6 +507,8 @@ def _check_type(self, value, append=False, cfg=None):
                         val = adapt_typehints(val, self._typehint, **kwargs)
                 except ValueError as ex:
                     assert ex  # needed due to ruff bug that removes " as ex"
+                    if orig_val == "-":
+                        raise ex
```
[the remark this answers is not in the document]
Comment (mquillot) [reply jsonargparse/_typehints.py:511]: Ok, I let you propose a modification?

File: jsonargparse/_util.py [hunk 4]
```diff
@@ -653,7 +667,12 @@ def get_content(self, mode: str = "r") -> str:
     @contextmanager
     def open(self, mode: str = "r") -> Iterator[IO]:
         """Return an opened file object for the path."""
-        if self._is_url:
+        if self.std_io:
+            if "r" in mode:
+                yield sys.stdin
+            elif "w" in mode:
+                yield sys.stdout
```
[the remark this answers is not in the document]
Comment (mquillot) [reply jsonargparse/_util.py]: I let you propose a modification?

[the remark this answers is not in the document]
Comment (mauvilsa) [reply jsonargparse/_util.py]: No proposal. I just noticed this. But I am not sure how much of a problem would it be. Lets just leave it like this and only do something about it if it creates a problem for someone.

File: jsonargparse/_typehints.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mauvilsa) [reply jsonargparse/_typehints.py:511]: I don't yet know. We can leave it like this here, but will take a note to look at this later.

**Review by mauvilsa — APPROVED** [review]
Looks good!

**Pull request merged**